### PR TITLE
double negation case

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Add some special cases involving negation

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: tests/unit/test_expression_tree/test_unary_operators.py [hunk 1]
```diff
@@ -190,6 +190,12 @@ def test_div(self):
         div = pybamm.div(-pybamm.Gradient(a))
         self.assertEqual(div.id, (-pybamm.Divergence(pybamm.Gradient(a))).id)
 
+        div = pybamm.div(-a * pybamm.Gradient(a))
+        self.assertEqual(div.id, (-pybamm.Divergence(a * pybamm.Gradient(a))).id)
+
+        # div = pybamm.div(a * -pybamm.Gradient(a))
```
Comment (brosaplanella) [inline tests/unit/test_expression_tree/test_unary_operators.py:196]: Why is this commented out?

**Review by brosaplanella — APPROVED** [review]
Looks good Tino, thanks! Just check the comment on the commented lines.

File: tests/unit/test_expression_tree/test_unary_operators.py (hunk 1 above)
Comment (tinosulzer) [reply tests/unit/test_expression_tree/test_unary_operators.py:196]: it causes errors that I couldn't fix

Comment (tinosulzer) [reply tests/unit/test_expression_tree/test_unary_operators.py:196]: not a crucial line

**Pull request merged**